adam/math_3d.py

```diff
@@ -22,3 +22,30 @@ class Point:
     z: float = attrib(  # pylint:disable=invalid-name
         validator=instance_of(float), converter=float
     )
+
+
+@attrs(frozen=True)
+class DepthPoint:
+    """
+    A point in 3D space represented by a depth value for the z direction.
+
+    Points are view agnostic by implementation but are assumed to exist
+    from the space view-point when compared. Our system may use multiple views
+    at different points which are not 1:1 comparable.
```

Comment:
Nit: "the space viewpoint" sounds to me like there is necessarily just one. So:
```suggestion
    Points are view agnostic by implementation but are assumed to exist
    from some specific view-point in space when compared. Our system may use multiple views
    at different points which are not 1:1 comparable.
```

I am happy with the second sentence here though, that gets the main point across.